networkit/cpp/centrality/Betweenness.cpp

```diff
@@ -59,68 +48,53 @@ void Betweenness::run() {
+        sssp.setSource(s);
         if (!handler.isRunning()) return;
-        sssp->run();
+        sssp.run();
         if (!handler.isRunning()) return;
         // compute dependencies for nodes in order of decreasing distance from s
-        std::vector<node> stack = sssp->getNodesSortedByDistance();
+        std::vector<node> stack = sssp.getNodesSortedByDistance();
         while (!stack.empty()) {
             node t = stack.back();
             stack.pop_back();
-            for (node p : sssp->getPredecessors(t)) {
+            for (node p : sssp.getPredecessors(t)) {
                 // workaround for integer overflow in large graphs
-                bigfloat tmp = sssp->numberOfPaths(p) / sssp->numberOfPaths(t);
+                bigfloat tmp = sssp.numberOfPaths(p) / sssp.numberOfPaths(t);
                 double weight;
                 tmp.ToDouble(weight);
                 double c= weight * (1 + dependency[t]);
                 dependency[p] += c;
+
                 if (computeEdgeCentrality) {
-                    edgeScorePerThread[omp_get_thread_num()][G.edgeId(p,t)] += c;
+                    const edgeid edgeId = G.edgeId(p, t);
+#pragma omp atomic
+                    edgeScoreData[edgeId] += c;
                 }
-
-
-            }
-            if (t != s) {
-                scorePerThread[omp_get_thread_num()][t] += dependency[t];
             }
+
+            if (t != s)
+#pragma omp atomic
+                scoreData[t] += dependency[t];
         }
     };
     handler.assureRunning();
     G.balancedParallelForNodes(computeDependencies);
     handler.assureRunning();
-    DEBUG("adding thread-local scores");
-    // add up all thread-local values
-    for (const auto &local : scorePerThread) {
-        G.parallelForNodes([&](node v){
-            scoreData[v] += local[v];
-        });
-    }
-    if (computeEdgeCentrality) {
-        for (const auto &local : edgeScorePerThread) {
-            for (count i = 0; i < local.size(); i++) {
-                edgeScoreData[i] += local[i];
-            }
-        }
-    }
+
     if (normalized) {
         // divide by the number of possible pairs
         count n = G.numberOfNodes();
         count pairs = (n-2) * (n-1);
         count edges =  n    * (n-1);
-        if (!G.isDirected()) {
-            pairs = pairs / 2;
-            edges = edges / 2;
-        }
-        G.forNodes([&](node u){
+        G.parallelForNodes([&](node u){
             scoreData[u] = scoreData[u] / pairs;
         });
-        if (computeEdgeCentrality) {
-            for (count edge = 0; edge < edgeScoreData.size(); edge++) {
-                edgeScoreData[edge] =  edgeScoreData[edge] / edges;
-            }
-        }
+
+        if (computeEdgeCentrality)
+#pragma omp parallel for
+            for (omp_index i = 0; i < static_cast<omp_index>(edgeScoreData.size()); ++i)
+                edgeScoreData[i] =  edgeScoreData[i] / static_cast<double>(edges);
```

Comment:
For the sake of readability I would suggest to add some braces around these code blocks, in particular with the `#pragma` that is not indented this looks weird.

Reply:
This block might look suspicious indeed, I added braces.